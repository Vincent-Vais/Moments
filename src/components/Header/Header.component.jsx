import React from "react";

import Auth from "../Auth";

import logo from "../../assets/logo.svg";

function Header({ classes }) {
  return (
    <div className={classes.header}>
      <h4 className={classes.header__name}>moments</h4>
      <img src={logo} alt="logo" className={classes.header__logo} />
      <Auth />
    </div>
  );
}

export default Header;
