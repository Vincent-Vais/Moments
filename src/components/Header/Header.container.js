import React from "react";
import useStyles from "./Header.styles";

function Container({ Component, ...otherProps }) {
  const classes = useStyles();
  return <Component classes={classes} {...otherProps} />;
}

export default (component) => (props) => (
  <Container Component={component} {...props} />
);
