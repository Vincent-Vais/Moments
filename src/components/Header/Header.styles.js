import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  header: {
    background: `${theme.palette.bcg.main}`,
    padding: "0 2rem",
    borderBottom: `1px solid ${theme.palette.primary.main}`,
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
  },
  header__name: {
    fontFamily: "Sansita Swashed, cursive",
    fontSize: "2.5rem",
    color: `${theme.palette.text.main}`,
    letterSpacing: "1px",
  },
  header__logo: {
    objectFit: "contain",
    width: "7.5rem",
  },
}));

export default useStyles;
