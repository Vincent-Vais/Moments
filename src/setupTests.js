import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";
import { createMuiTheme } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core/styles";

configure({ adapter: new Adapter() });

export function MockTheme({ children }) {
  const theme = createMuiTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

const middlewares = [];
export const mockStore = configureStore(middlewares);

export const mockClasses = {};
