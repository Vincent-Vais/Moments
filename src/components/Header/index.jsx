import Header from "./Header.component";
import WithData from "./Header.container";

export default WithData(Header);
