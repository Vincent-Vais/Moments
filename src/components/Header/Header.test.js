import React from "react";
import { shallow } from "enzyme";
import Header from "./Header.component";
import { mockClasses } from "../../setupTests";

const getPresentComponent = () => <Header classes={mockClasses} />;

describe("Header component", () => {
  it("renders presentational component", () => {
    const component = shallow(getPresentComponent());
    expect(component).toMatchSnapshot();
  });
});
