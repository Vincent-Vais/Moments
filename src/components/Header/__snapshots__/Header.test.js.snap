// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders presentational component 1`] = `
<div>
  <h4>
    moments
  </h4>
  <img
    alt="logo"
    src="logo.svg"
  />
  <Auth />
</div>
`;
